Tests broken with flask-2.3.2

After upgrading Flask to 2.3.2, the tests fail to collect:

```pytb
========================================================= test session starts =========================================================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
rootdir: /tmp/flask-api
collected 6 items / 6 errors                                                                                                          

=============================================================== ERRORS ================================================================
____________________________________________ ERROR collecting flask_api/tests/test_app.py _____________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_app.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
_________________________________________ ERROR collecting flask_api/tests/test_exceptions.py _________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_exceptions.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
_________________________________________ ERROR collecting flask_api/tests/test_mediatypes.py _________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_mediatypes.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
________________________________________ ERROR collecting flask_api/tests/test_negotiation.py _________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_negotiation.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
__________________________________________ ERROR collecting flask_api/tests/test_parsers.py ___________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_parsers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
_________________________________________ ERROR collecting flask_api/tests/test_renderers.py __________________________________________
ImportError while importing test module '/tmp/flask-api/flask_api/tests/test_renderers.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
flask_api/tests/__init__.py:14: in <module>
    exec("from flask_api.tests.%s import *" % module)
flask_api/tests/test_renderers.py:4: in <module>
    from flask.json import JSONEncoder
E   ImportError: cannot import name 'JSONEncoder' from 'flask.json' (/tmp/flask-api/.venv/lib/python3.11/site-packages/flask/json/__init__.py)
========================================================== warnings summary ===========================================================
flask_api/renderers.py:6
  /tmp/flask-api/flask_api/renderers.py:6: DeprecationWarning: '_request_ctx_stack' is deprecated and will be removed in Flask 2.4.
    from flask.globals import _request_ctx_stack

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================= short test summary info =======================================================
ERROR flask_api/tests/test_app.py
ERROR flask_api/tests/test_exceptions.py
ERROR flask_api/tests/test_mediatypes.py
ERROR flask_api/tests/test_negotiation.py
ERROR flask_api/tests/test_parsers.py
ERROR flask_api/tests/test_renderers.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 6 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================== 1 warning, 6 errors in 0.77s =====================================================
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,7 +15,7 @@ env:
   global:
     - RANDOM_SEED=0
   matrix:
-    - FLASK_VERSION=2.0.1
+    - FLASK_VERSION=2.3.2
 
 before_install:
   - pip install pipenv

--- a/flask_api/renderers.py
+++ b/flask_api/renderers.py
@@ -60,8 +60,8 @@ class JSONRenderer(BaseRenderer):
             indent = None
         # Indent may be set explicitly, eg when rendered by the browsable API.
         indent = options.get("indent", indent)
-        return json.dumps(
-            data, cls=current_app.json_encoder, ensure_ascii=False, indent=indent
+        return current_app.json.dumps(
+            data, ensure_ascii=False, indent=indent
         )
 
 

--- a/flask_api/renderers.py
+++ b/flask_api/renderers.py
@@ -1,4 +1,3 @@
-import json
 import pydoc
 import re
 

--- a/flask_api/tests/test_renderers.py
+++ b/flask_api/tests/test_renderers.py
@@ -1,7 +1,7 @@
 import unittest
 from datetime import datetime
 
-from flask.json import JSONEncoder
+from flask.json.provider import DefaultJSONProvider
 
 from flask_api import FlaskAPI, renderers, status
 from flask_api.decorators import set_renderers
@@ -39,14 +39,14 @@ class RendererTests(unittest.TestCase):
         self.assertEqual(content, expected)
 
     def test_render_json_with_custom_encoder(self):
-        class CustomJsonEncoder(JSONEncoder):
+        class CustomJsonProvider(DefaultJSONProvider):
             def default(self, o):
                 if isinstance(o, datetime):
                     return o.isoformat()
                 return super().default(o)
 
         app = self._make_app()
-        app.json_encoder = CustomJsonEncoder
+        app.json = CustomJsonProvider(app)
         renderer = renderers.JSONRenderer()
         date = datetime(2017, 10, 5, 15, 22)
         with app.app_context():

